src/index.ts

```diff
@@ -26,17 +27,23 @@ function managerEntries(entry: string[] = []) {
 // Uses a recursive setTimeout instead of setInterval to avoid overlapping async calls.
 const observeGitInfo = async (
   interval: number,
-  callback: (info: GitInfo, prevInfo: GitInfo) => void
+  callback: (info: GitInfo, prevInfo: GitInfo) => void,
+  errorCallback: (e: Error) => void
 ) => {
   let prev: GitInfo;
   let timer: NodeJS.Timeout | undefined;
   const act = async () => {
-    const gitInfo = await getGitInfo();
-    if (Object.entries(gitInfo).some(([key, value]) => prev?.[key as keyof GitInfo] !== value)) {
-      callback(gitInfo, prev);
+    try {
+      const gitInfo = await getGitInfo();
+      if (Object.entries(gitInfo).some(([key, value]) => prev?.[key as keyof GitInfo] !== value)) {
+        callback(gitInfo, prev);
+      }
+      prev = gitInfo;
+      timer = setTimeout(act, interval);
+    } catch (e: any) {
+      console.error("getGitInfo failed. Ending loop.", e);
```

Comment:
```suggestion
      console.error(`Failed to fetch git info, with error:\n${e}`);
```

I'm not sure the stack trace is really all that helpful.